cms/djangoapps/contentstore/views/tests/test_block.py

```diff
@@ -3473,3 +3497,20 @@ def test_self_paced_item_visibility_state(self, store_type):
         # Check that in self paced course content has live state now
         xblock_info = self._get_xblock_info(chapter.location)
         self._verify_visibility_state(xblock_info, VisibilityState.live)
+
+
+class TestUpdateFromSource(MixedSplitTestCase):
+    """
+    Test update_from_source.
+    """
+
+    def test_update_from_source(self):
+        """
+        Test that update_from_source updates the destination block.
+        """
+        course = CourseFactory()
+        user = UserFactory.create()
+        source_chapter = BlockFactory(parent=course, category='chapter', display_name='Source Chapter')
+        destination_chapter = BlockFactory(parent=course, category='chapter', display_name='Destination Chapter')
+        update_from_source(source_block=source_chapter, destination_block=destination_chapter, user_id=user.id)
+        self.assertEqual(self.store.get_item(destination_chapter.location).display_name, 'Source Chapter')
```

Comment:
Would it make sense to verify that the `metadata`, `definition_data` and `asides` are updated correctly too?

Reply:
This turned out to be harder to do than I expected, but I've added tests for this now as well as a variant that verifies that they get properly clobbered as they should be if content exists there already.